lib/macho/exceptions.rb

```diff
@@ -3,6 +3,10 @@ module MachO
   class MachOError < RuntimeError
   end
 
+  # A generic Mach-O error during file modification.
+  class MachOModificationError < MachOError
```

Comment:
Let's be super-explicit here. I think this should be something like “A generic Mach-O error during file modification, that is non-fatal and might be suppressed if operating in non-strict mode.” (I'm not particularly happy with this wording, but I hope the message gets across. Please feel free to phrase this better!)

We could also consider a name different from `MachOModificationError` that better reflects the nature of these exceptions.